crates/cheatcodes/spec/src/vm.rs

```diff
@@ -825,6 +825,30 @@ interface Vm {
     #[cheatcode(group = Testing, safety = Safe)]
     function assumeNoRevert() external pure;
 
+    /// Discard this run's fuzz inputs and generate new ones if next call reverted with the given revert data. Call more than once to add multiple reasons. Can be combined with `assumeNoPartialRevert`.
+    #[cheatcode(group = Testing, safety = Safe)]
+    function assumeNoRevert(bytes4 revertData) external pure;
+
+    /// Discard this run's fuzz inputs and generate new ones if next call reverted with the given revert data. Call more than once to add multiple reasons. Can be combined with `assumeNoPartialRevert`.
+    #[cheatcode(group = Testing, safety = Safe)]
+    function assumeNoRevert(bytes calldata revertData) external pure;
+
+    /// Discard this run's fuzz inputs and generate new ones if next call reverted with the given revert data. Call more than once to add multiple reasons. Can be combined with `assumeNoRevert`.
+    #[cheatcode(group = Testing, safety = Safe)]
+    function assumeNoRevert(bytes4 revertData, address reverter) external pure;
+
+    /// Discard this run's fuzz inputs and generate new ones if next call reverted with the given revert data. Call more than once to add multiple reasons. Can be combined with `assumeNoPartialRevert`.
+    #[cheatcode(group = Testing, safety = Safe)]
+    function assumeNoRevert(bytes calldata revertData, address reverter) external pure;
+
+    /// Discard this run's fuzz inputs and generate new ones if next call reverted with data starting with the given revert data. Call more than once to add multiple reasons. Can be combined with `assumeNoRevert`.
```

Comment:
I think we can get rid of `function assumeNoPartialRevert` cheatcodes, the `assumeParatialRevert` counterparts were added to avoid breaking changes but should not be the case for these new cheatcods

Reply:
@grandizzy just to make sure I understand – are you saying that `vm.assumeNoRevert(bytes4)` should behave like `assumeNoPartialRevert(bytes4)`?

Reply:
Yeah, that's my take, wdyt?

Reply:
i like it! makes it cleaner.